### DISCUSSION TITLE
Checking for dead urls

### DISCUSSION THREAD

**Discussion by michellaurent** [maizzle/framework#764] [Ideas]
I'm very happy with the power of Maizzle to build HTML emails. 

I would be nice to be able to validate that all urls entered in a link (href) are correct/reachable (url returns a "valid" HTTP code).

Any idea how to do this ?

- Create a transformer that will add a ❌ next to the unreachable url ?
- Throws an error if a link url is unreachable (add a function to do that in the `afterRender` event) ?
- ?

**Comment by michellaurent** [comment]
Maybe this article could help:

[https://silvenon.com/blog/checking-for-dead-urls](https://silvenon.com/blog/checking-for-dead-urls)

**Comment by cossssmin** [comment]
For local dev in your editor I'd probably go with something like [this](https://marketplace.visualstudio.com/items?itemName=BillDietrich.linkcheckerhtml), otherwise yeah `afterRender` would be a good spot. You could probably just get away with using something like `axios` in `afterRender` and validating that you got a `200` back 👍

**Reply by michellaurent** [reply]
I can check the status of all links in the html content using `axios` but how can I return an error from this `afterRender` event to the build script ?
